packages/gatsby-plugin-mdx/src/compile-mdx.ts

```diff
@@ -0,0 +1,27 @@
+import type { ProcessorOptions } from "@mdx-js/mdx"
+import type { IFileNode, IMdxNode } from "./types"
+
+// Compiles MDX into JS
+// This could be replaced by @mdx-js/mdx if MDX compile would
+// accept custom data passed to the unified pipeline via processor.data()
+export default async function compileMDX(
+  source: string,
+  mdxNode: IMdxNode,
+  fileNode: IFileNode,
+  options: ProcessorOptions
+): Promise<string> {
+  const { createProcessor } = await import(`@mdx-js/mdx`)
+  const { VFile } = await import(`vfile`)
+
+  const processor = createProcessor(options)
+
+  // If we could pass this via MDX loader config, this whole custom loader might be obsolete.
+  processor.data(`mdxNode`, mdxNode)
+
+  const result = await processor.process(
+    // Inject path to original file for remark plugins. See: https://github.com/gatsbyjs/gatsby/issues/26914
```

Comment:
Do we really still need that?

Reply:
When I look at https://unifiedjs.com/learn/guide/create-a-plugin/ the `file` as a second argument is still there. So there will be remark plugins that rely on that.

Isn't the official loader also doing this?

Reply:
At least not in the way our test is checking it:

https://github.com/gatsbyjs/gatsby/blob/feat/mdx-v2-plugin-support/e2e-tests/mdx/gatsby-config.js#L39-L49

Reply:
@axe312ger then we'll need to update our test later